include/envoy/http/header_map.h

```diff
@@ -352,6 +352,9 @@ class HeaderEntry {
   virtual const HeaderEntry* name() const PURE;                                                    \
   virtual HeaderEntry* name() PURE;                                                                \
   virtual HeaderEntry& insert##name() PURE;                                                        \
+  virtual void setReference##name(const std::string& value) PURE;                                  \
```

Comment:
Done. The `HeaderMapImpl::setReference` methods also take a `const std::string&` as well, since they call the `HeaderString(const std::string&)` constructor. Worth changing over?

Reply:
yeah, fine with just a TODO either. `const std::string&` -> `absl::string_view` is implicitly convertible but not vice versa.